### ISSUE TITLE
ArrayOutOfBoundsException in vcf2adam (spark2_2.11-0.22.0) on UK10K VCFs (VCFv4.1)

### ISSUE THREAD

**Issue by Veryku** [bigdatagenomics/adam#1494]
Running vcf2adam on UK10K VCF files fails on an ArrayOutOfBoundsException midway through the run. Here's the call I used:
```
./bin/adam-submit --master yarn-client --num-executors 100 --executor-cores 4 \
--executor-memory 32g --driver-memory 32g -- \
vcf2adam /data/UK10K/EGAD00001000740 /data/UK10K/bigADAMTest/results
```
and the resultant exception [here](https://gist.github.com/Veryku/22bc5c13785be8f35f6aad16d418e3d7)

Running with `-stringency LENIENT` allowed the run to continue without the exception appearing.

**Comment by fnothaft** [comment]
Hi @Veryku! Thanks for posting the exception. What version/commit of ADAM did this occur on?

**Comment by Veryku** [comment]
This was runnable version `adam-distribution-spark2_2.11-0.22.0/`

**Comment by Veryku** [comment]
I also just remembered that when running ADAM on an older version (`adam-core_2.10-0.19.0`) through Gnocchi, vcf2adam with the same call works. Taner has a hunch it may have something to do with the updated annotations logic.

**Comment by heuermh** [comment]
It is hard to follow the stack trace; maybe a Number=G genotype field (GL, for example) had the wrong number of values?

**Comment by fnothaft** [comment]
Pinging this. @Veryku do you have any updates?

**Comment by Veryku** [comment]
Unfortunately no, I haven't been dealing with VCF2ADAM in the recent weeks, as processing the UK10K data on lenient stringency provided the results I needed at the time.

**Comment by fnothaft** [comment]
Sounds good. I am going to close this for now, but please reopen if it reappears.
